### PR TITLE
Add Kill Node Modules extension.

### PR DESCRIPTION
**Pull request by NicolasMontone**
## Description

I want to delete the node_modules from my computer and I find a lot of terminal-base tools but I want it to do it via raycas

## Screencast

image
image

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by NicolasMontone** [comment]
Locally the lint works, does anyone have this problem?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- You can use `accessories` instead for the date to save space. You can add: `accessories={[{ date: item.lastModified, tooltip: item.lastModified.toLocaleString() }]}`

-  The trash icon in the list isn't working well, so would it be better to implement some kind of multiselect instead?

- Add `style={Action.Style.Destructive}` to delete action

There might be more after this round of feedback 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by NicolasMontone** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * You can use `accessories` instead for the date to save space. You can add: `accessories={[{ date: item.lastModified, tooltip: item.lastModified.toLocaleString() }]}`
> * The trash icon in the list isn't working well, so would it be better to implement some kind of multiselect instead?
> * Add `style={Action.Style.Destructive}` to delete action
> 
> There might be more after this round of feedback 🙂
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

> > The trash icon in the list isn't working well, so would it be better to implement some kind of multiselect instead?

I think it makes total sense to change the Trash icon, maybe the folder Icon could work, right?

File: extensions/kill-node-modules/src/utils/format.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function formatSize(bytes: number): string {
+  const units = ["B", "KB", "MB", "GB", "TB"];
+  let size = bytes;
+  let unitIndex = 0;
+  while (size >= 1024 && unitIndex < units.length - 1) {
+    size /= 1024;
+    unitIndex++;
+  }
+  return `${size.toFixed(2)} ${units[unitIndex]}`;
+}
```
Comment (pernielsentikaer) [inline extensions/kill-node-modules/src/utils/format.ts:10]:
```suggestion
export function formatSize(bytes: number): string {
  const units = ["B", "KB", "MB", "GB", "TB"];
  let size = bytes;
  let unitIndex = Math.floor(Math.log(size) / Math.log(1024));
  size /= Math.pow(1024, unitIndex);
  return `${size.toFixed(2)} ${units[unitIndex]}`;
}
```

File: extensions/kill-node-modules/src/kill-node-modules.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  size: number;
+};
+
+export default function Command() {
+  const { items, loading, pagination } = useNodeModules();
+  const [deletedItems, setDeletedItems] = useState<Set<string>>(new Set());
+  const [deletingItems, setDeletingItems] = useState<Set<string>>(new Set());
+  const [sortWith, setSortWith] = useState<"size" | "lastModified">("lastModified");
+
+  const handleDelete = useCallback(async (id: string, path: string) => {
```
Comment (pernielsentikaer) [inline extensions/kill-node-modules/src/kill-node-modules.tsx:25]: Should we add a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) with       `rememberChoice: true`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Is the search too fuzzy?

Raycast03092024-fvMdv6gi png@2x

File: extensions/kill-node-modules/src/utils/format.ts (hunk 1 above)
Comment (NicolasMontone) [reply extensions/kill-node-modules/src/utils/format.ts:10]: Totally right!

**Comment by NicolasMontone** [comment]
> Is the search too fuzzy?
> 
> Raycast03092024-fvMdv6gi png@2x

You think so? I think is ok since maybe I want to check node_modules from a specific subfolder ie. /path/to/some/folder/[I want to search this]/...

I can implement another type of search, any suggestion?

**Comment by NicolasMontone** [comment]
@pernielsentikaer what do you think about merging this extension? Do you think that it is something missing?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by NicolasMontone** [comment]
> Hi 👋 
> 
> 
> 
> Looks good to me, approved 🔥

Thanks a lot for all the feedback and your time!

**Pull request merged**